### PR TITLE
plugin/cache: update comment to conform to the implementation

### PR DESCRIPTION
**Pull request by xiez**
### 1. Why is this pull request needed and what does it do?

Update comment accordingly to the code fix in https://github.com/coredns/coredns/commit/159a8fb68c44b238ccf603e9d7bd822519dcfb41 :wink: 

### 2. Which issues (if any) are related?

no

### 3. Which documentation changes (if any) need to be made?

no

### 4. Does this introduce a backward incompatible change or deprecation?

no

### CODE REVIEW

File: plugin/cache/item.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func newItem(m *dns.Msg, now time.Time, d time.Duration) *item {
 }
 
 // toMsg turns i into a message, it tailors the reply to m.
-// The Authoritative bit is always set to 0, because the answer is from the cache.
```
Comment (miekg) [inline plugin/cache/item.go:52]: good catch. Please make this comment longer. 

The Authoritative bit should be set to 0, but some client stub resolver implementations, most notably, the one from Python will discard answers that do not have this bit set. So we're forced to always set this to 1; regardless if the answer came from the cache or not.

Comment (xiez) [reply plugin/cache/item.go:52]: updated. 😄 

Just a little FYI, Python use low-level glibc function `getaddrinfo`  (https://github.com/python/cpython/blob/3.6/Modules/socketmodule.c#L6061) to perform address resolution. On some legacy systems(e.g. ubuntu 14.04 with glib version 2.20), this function will discard answers that do not have Authoritative bit set, which means not only Python, other languages like ruby/C also suffer the same issue;  On newer systems(e.g. ubuntu 16.04 with glib version 2.23), this issue is resolved.

Comment (miekg) [reply plugin/cache/item.go:52]: Can you add that too; I don't mind that we have a long comment here capturing all these bits.

Comment (xiez) [reply plugin/cache/item.go:52]: ok, how about this:

```
// The Authoritative bit should be set to 0, but some client stub resolver implementations, most notably,
// on some legacy systems(e.g. ubuntu 14.04 with glib version 2.20), low-level glibc function `getaddrinfo`
// useb by Python/Ruby/etc.. will discard answers that do not have this bit set.
// So we're forced to always set this to 1; regardless if the answer came from the cache or not.
// On newer systems(e.g. ubuntu 16.04 with glib version 2.23), this issue is resolved.
// So we may set this bit back to 0 in the future ?
```

**Comment by miekg** [comment]
>ok, how about this:
>
>```
>// The Authoritative bit should be set to 0, but some client stub resolver implementations, most notably,
>// on some legacy systems(e.g. ubuntu 14.04 with glib version 2.20), low-level glibc function `getaddrinfo`
>// useb by Python/Ruby/etc.. will discard answers that do not have this bit set.
>// So we're forced to always set this to 1; regardless if the answer came from the cache or not.
>// On newer systems(e.g. ubuntu 16.04 with glib version 2.23), this issue is resolved.
>// So we may set this bit back to 0 in the future ?
>```

LGTM!

**Pull request merged**